bescort.lic

```diff
@@ -966,28 +966,46 @@ class Bescort
 
     path = mode == 'shard' ? $ICE_PATH_SHARD : $ICE_PATH_HIB
 
-    if exists?('skates')
-      EquipmentManager.new.empty_hands
-      footwear = get_settings.footwear
-      fput("remove #{footwear}") if footwear
-      bput("stow my #{footwear}", 'You put') if footwear
-      fput('get skates')
-      fput('wear skates')
-      path.each do |movement|
-        move(movement)
+    EquipmentManager.new.empty_hands
+    footwear = get_settings.footwear
+    move_fast = false
+
+    if DRCI.exists?("skates")
+      if !footwear && DRCI.get_item?("skates") && DRCI.wear_item?("skates")
+        move_fast = true
+      elsif footwear && DRCI.remove_item?(footwear)
```

Comment:
If I'm reading this right, then we remove the footwear here. If any of the next steps (put away footwear, get skates, wear skates) fail, we never wear the footwear again (since `move_fast` is never set to true). Is that right?

Reply:
Actually, I thought this would be caught by the else, or do I need a local else here?

Reply:
If `footwear` isn't defined _OR_ removing it fails, then control moves to the next `elsif` (line 982). However, once those two succeed and control moves into _THIS_ `elsif` (line 976), control will never go into the other `elsif` or `else` blocks. They're all mutually exclusive.

To do what you're describing, we could add the conditionals on lines 977 and 978 to the `elsif` on line 976 (and make a similar refactor to the other `elseif` with a nested `if` on line 982). If we're relying on the `else`, then the cleanup there needs to be very flexible/general to handle the different states we could be in.

Reply:
I ended up testing, and I see what you mean now. I think the current version catches oddities well.

Reply:
I still think this isn't catching all of the cases. For example, on line 978 if getting skates succeeds but wearing them fails, then we never set `move_fast` to true (which means the cleanup on line 1002 never happens). But we still have skates in our hands.

Reply:
> I don't think I have a preference as long as both handle each state.

I thought about this more. Stylistically, I have a strong preference for the merged `if`s with a single general-purpose recovery/cleanup `else`. I'm a bit concerned about the approach though, because my gut says there could be edge cases (what if the user has two boots with the same name?).

Reply:
Let me think on it more. 

The goal here was to not die on the road, which this achieves now. 

I wouldn't worry about two boots named the same, because that problem exists today. If we keep that specific problem, but make the overall situation better, I'd call that progress. Does that sound reasonable?